client/core/trade.go

```diff
@@ -533,7 +580,21 @@ func (t *trackedTrade) tick() (int, error) {
 			t.notify(newOrderNote("Match complete", details, db.Poke, corder))
 		}
 	}
-	return len(swaps) + len(redeems), errs.ifany()
+
+	if len(refunds) > 0 {
+		refunded, err := t.refundMatches(refunds)
+		corder, _ := t.coreOrderInternal()
+		details := fmt.Sprintf("Refunded %.8f %s on order %s",
+			float64(refunded)/conversionFactor, unbip(t.wallets.fromAsset.ID), t.token())
+		if err != nil {
+			errs.addErr(err)
+			t.notify(newOrderNote("Match failure", details+", with some errors", db.ErrorLevel, corder))
+		} else {
+			t.notify(newOrderNote("Match failure", details, db.WarningLevel, corder))
+		}
+	}
```

Comment:
I think this would be better as two notifications. One for the match failure, and possibly another if there is a refund error.